### PR TITLE
Turned return type annotations of private methods into php return types

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

As discussed with @nicolas-grekas in https://github.com/symfony/symfony/pull/30323#issuecomment-518638124, this PR attempts to turn `@return` annotations on private methods into return type declarations.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php [hunk 1]
```diff
@@ -188,8 +188,6 @@ protected function describeContainerEnvVars(array $envs, array $options = [])
 
     /**
      * Writes data as json.
-     *
-     * @return array|string
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php]: I'm pretty certain that this method does not return anything.

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php]: 3.4 or lower?

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php [hunk 2]
```diff
@@ -395,7 +395,7 @@ protected function describeCallable($callable, array $options = [])
     /**
      * @return string
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]: legacy

File: src/Symfony/Component/Console/Terminal.php [hunk 3]
```diff
@@ -106,17 +106,17 @@ private static function getConsoleMode()
...
      * Runs and parses stty -a if it's available, suppressing any error output.
-     *
-     * @return string|null
      */
-    private static function getSttyColumns()
+    private static function getSttyColumns(): ?string
     {
         if (!\function_exists('proc_open')) {
-            return;
+            return null;
```
Comment (derrabus) [inline src/Symfony/Component/Console/Terminal.php]: After adding the return type, php will error if I don't explicitly return `null`.

**Comment by derrabus** [comment]
I'll fix broken PHPDocs and inconsistent returns that I've found while working on this PR with separate PRs on lower branches, probably tomorrow.

File: src/Symfony/Component/Console/Terminal.php (hunk 3 above)
Comment (Tobion) [reply src/Symfony/Component/Console/Terminal.php]: ref #17201

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 4]
```diff
@@ -149,7 +149,7 @@ private function parseDefinitions(\DOMDocument $xml, string $file, array $defaul
      *
      * @return array
```
Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: forgot to remove this one

File: src/Symfony/Component/HttpKernel/HttpCache/Store.php [hunk 5]
```diff
@@ -341,7 +335,7 @@ private function doPurge(string $url)
     /**
      * Loads data for the given key.
      *
-     * @return string The data associated with the key
+     * @return string|false The data associated with the key
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/HttpCache/Store.php]: IMO we should change this to a nullable string return value

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 6]
```diff
@@ -770,13 +762,8 @@ private function camelize(string $string): string
 
     /**
      * Searches for add and remove methods.
-     *
-     * @param \ReflectionClass $reflClass The reflection class for the given object
-     * @param array            $singulars The singular form of the property name or null
-     *
-     * @return array|null An array containing the adder and remover when found, null otherwise
```
Comment (Tobion) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php]: alot of wrong phpdocs here ^^

File: src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php [hunk 7]
```diff
@@ -165,11 +163,9 @@ private function attemptSwitchUser(Request $request, string $username)
     /**
      * Attempts to exit from an already switched user.
      *
-     * @return TokenInterface The original TokenInterface instance
```
Comment (Tobion) [inline src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php]: the description seems value to be kept. maybe as a normal description with `@return`

File: src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php [hunk 8]
```diff
@@ -114,12 +114,10 @@ public function __invoke(RequestEvent $event)
     /**
      * Attempts to switch to another user.
      *
-     * @return TokenInterface|null The new TokenInterface if successfully switched, null otherwise
```
Comment (Tobion) [inline src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php]: the description seems value to be kept. maybe as a normal description with `@return`

**Comment by nicolas-grekas** [comment]
can be rebased

**Comment by derrabus** [comment]
@nicolas-grekas The PR will probably conflict again once #33007 is merged, so I'll wait for that.

File: src/Symfony/Component/HttpKernel/HttpCache/Store.php (hunk 5 above)
Comment (derrabus) [reply src/Symfony/Component/HttpKernel/HttpCache/Store.php]: Fixed in #33007.

**Comment by derrabus** [comment]
Done.

**Comment by nicolas-grekas** [comment]
(tests fail)

**Comment by derrabus** [comment]
Fixed.

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**